### DISCUSSION TITLE
a more logical bookmark button order?

### DISCUSSION THREAD

**Discussion by axd1967** [Stellarium/stellarium#1812] [Ideas]
The order of buttons in the bookmark dialog has in fact always been frustrating me. I consider myself being an experienced Stellarium user, yet find myself often - if not always - having to find the correct button in the bookmark dialog.

This is the current (0.21) order:
![image](https://user-images.githubusercontent.com/3529789/126304876-5b4d421a-4d31-4714-8302-0782fe8b707c.png)

It has a few issues that keep rubbing against my ergonomical nose. The issue is amplified by the poor choice of UI colour (see also #678 ) that makes it harder to find the correct button.

Here is a proposed change. _The main idea is that there is a tighter visual (vertical) binding between similar buttons that represent opposing actions._

![image](https://user-images.githubusercontent.com/3529789/126306299-75242ce0-4374-422a-8297-eb1963cd37d0.png)

1. add/delete are the most used buttons and are in a more **logical up/down** order
2. the highlight/clear highlights buttons are more often used and are in a more logical up/down order
3. import/export are used less often, thus moved more to the right
3. the "go to" button is not so important, as it is the same as double-clicking a bookmark
2. the "clear bookmarks" button is dangerous and moved to the extreme right

Obviously there will be differing points of view, especially on frequency of use...

The biggest problem is always: can users accept change? (yes, if they are flexible, no if they are stuck in sometimes bad/wrong routines). The alternative is of course to fork off Stellarium.

What could be the impact of this change to someone who is "stuck" (and e.g. no longer watches the button texts, but clicks by habit, and therefore risks destroying things by mistake)?

2. the "stuck user" wants to import bookmarks but will now (inadvertently) invoke the "remove bookmark" action -> **this is problematic**. ideally, a confirmation dialog could guard for this issue, but the user would have to confirm every bookmark deleted. open for discussion (my point of view: yes, we need either a confirmation dialog OR an "undo" feature)
3. the "stuck user" wants to "highlight bookmarks" and will inadvertently activate "clear bookmarks". --> this is problematic, but the solution is trivial _and needs to be implemented anyway_: the "clear bookmarks" action needs to be protected with a confirmation dialog. ( https://github.com/Stellarium/stellarium/pull/181 -> GitHub bug, see PR181)
4. as far as I can see, the other changes will not cause irreversible damage

----

One tip to be aware of is that many Stellarium dialogs can be moved partially off-screen, exposing just a few important buttons. Here are some examples:

By pushing the settings/landscape (F4) dialog partially out of the screen one can quickly toggle (by double-clicking) landscapes as the buttons are just nicely visible:
![image](https://user-images.githubusercontent.com/3529789/126305402-6c356bd7-b61c-4907-9444-d640f4a33e8e.png)

Same for bookmarks, to quickly jump between bookmarks by double-clicking:

![image](https://user-images.githubusercontent.com/3529789/126307093-f2f036a6-fb27-4d95-ad8a-ba5b7f1d20d9.png)

By pushing most of the Sky tab outside the left side of the screen, one can play with the magnitude filter and immediately see the result:

![image](https://user-images.githubusercontent.com/3529789/126307274-040ae277-8036-42e5-900c-14eb5093904c.png)

Or if you want to see the effect of the Bortle setting, push the dialog out to the right:

![image](https://user-images.githubusercontent.com/3529789/126307141-99f4414c-387d-41af-aedc-7dd0ccae392a.png)

Or play with lat/long and immediately see the result:

![image](https://user-images.githubusercontent.com/3529789/126313710-2ebb8ca9-5e6e-4d38-b4d3-a75d3e1da64a.png)

**Comment by gzotti** [comment]
I usually move my dialogs out of the way, so this is an important feature. 
I can support a better arrangement of these buttons, but as I understand somebody is currently (?) working on extending bookmarks, this will probably void any proposals.
Protecting "delete all" by a confirmation dialog is OK. Confirming each single delete is IMO obtrusive. A bookmark can usually be restored from memory. 
As long as the program has not reached version 1.* [and even later], we can re-arrange buttons and rename properties as we wish, and blame the user for operating errors (as long as the button is correctly labeled :-).

**Comment by axd1967** [comment]
@gzotti I think you are referring to these issues?
- https://github.com/Stellarium/stellarium/issues/976 (and https://github.com/Stellarium/stellarium/pull/854?)
- https://github.com/Stellarium/stellarium/issues/100
